api/tools/src/main/java/org/pmiops/workbench/tools/elastic/BigQueryOmopIndexer.java

```diff
@@ -0,0 +1,325 @@
+          .map(line -> {
+            try {
+              return ElasticDocument.fromBigQueryJson(line);
+            } catch (IOException e) {
+              throw new RuntimeException("Error reading JSON line from BigQuery export", e);
+            }
+          })
+          .iterator();
+    } else {
+      log.info(
+          String.format("SELECTing from CDR '%s' -> Elasticsearch JSON documents", cdrDataset));
+      TableResult res = bq.query(QueryJobConfiguration.newBuilder(personSQL).build());
+      docs = StreamSupport.stream(res.iterateAll().spliterator(), false)
+          .map(ElasticDocument::fromBigQueryResults)
+          .iterator();
+    }
+
+    log.info(String.format("Starting bulk ingest of Elasticsearch documents to '%s'",
+        opts.getOptionValue(esBaseUrlOpt.getLongOpt())));
+    int fails = 0;
+    for (int i = 0; docs.hasNext(); i += BATCH_SIZE) {
+      BulkRequest bulkReq = new BulkRequest();
+      int j;
+      for (j = 0; j < BATCH_SIZE && docs.hasNext(); j++) {
+        ElasticDocument doc = docs.next();
+
+        // Cannot include _id in the document, needs to be part of the insertion request.
+        bulkReq.add(new IndexRequest(personIndex, ES_INDEX_TYPE, doc.id)
+            .source(doc.source));
+      }
+      BulkResponse response = client.bulk(bulkReq, ES_OPTS);
+      if (response.hasFailures()) {
+        for (BulkItemResponse itemResp : response.getItems()) {
+          if (itemResp.isFailed()) {
+            // TODO: Investigate retry handling or aggregate failures.
+            log.warning(itemResp.getFailureMessage());
+            fails++;
+          }
+        }
+      }
+
+      int inserted = i + j - fails;
+      log.info(String.format("Inserted %.2f%% of documents (%d/%d) (%d failed)",
+          ((float) 100 * (inserted)) / totalSampleSize, inserted, totalSampleSize, fails));
+    }
+  }
+
+  private int getPersonCount(BigQuery bq, String bqDataset, int inverseProb)
+      throws InterruptedException {
+    TableResult res = bq.query(QueryJobConfiguration.of(
+        String.format(
+            "SELECT SUM(1) FROM `%s.person` WHERE MOD(person_id, %d) = 0;",
+            bqDataset, inverseProb)));
+    return (int) Iterables.getOnlyElement(res.getValues()).get(0).getLongValue();
+  }
+
+  private String getPersonBigQuerySQL(String bqDataset, int inverseProb) throws IOException {
+    String esPersonBQTemplate = Resources.toString(
+        Resources.getResource("bigquery/es_person.sql"), Charset.defaultCharset());
+    return esPersonBQTemplate
+        .replaceAll("\\{BQ_DATASET\\}", bqDataset)
+        .replaceAll("\\{PERSON_ID_MOD\\}", Integer.toString(inverseProb));
+  }
+
+  private static RestHighLevelClient newClient(String esBaseUrl) throws MalformedURLException {
+    URL url = new URL(esBaseUrl);
+    return new RestHighLevelClient(RestClient.builder(
+        new HttpHost(url.getHost(), url.getPort(), url.getProtocol())));
+  }
+
+  @Bean
```

Comment:
Do we need to expose this as a Spring Bean?

Reply:
Tested - yes, this is what triggers the SpringApplicationBuilder to actually execute something.